test/test-apimatic-core/union_type_tests/one_of_test.rb

```diff
@@ -0,0 +1,1745 @@
+require 'minitest/autorun'
+require 'apimatic_core'
+
+require_relative '../../test-helper/models/morning'
+require_relative '../../test-helper/models/evening'
+require_relative '../../test-helper/models/month_number_enum'
+require_relative '../../test-helper/models/month_name_enum'
+
+class OneOfTest < Minitest::Test
```

Comment:
Question: Have we added unit tests for `nested OneOf cases`? If not, kindly add at-least 1 test case for that.